src/field/arena.ts

```diff
@@ -642,6 +642,31 @@ export class Arena {
     return TimeOfDay.DAWN;
   }
 
+  /**
+   * Sets a random weather based on the time of day and the current biome
+   */
+  setRandomWeather(): void {
+    const applySunBonus =
+      (this.getTimeOfDay() === TimeOfDay.DAWN || this.getTimeOfDay() === TimeOfDay.DAY)
+      && allBiomes.get(this.biomeType).sunBonus > 0;
+    const weatherPool = { ...allBiomes.get(this.biomeType).weatherPool };
+
+    // Applies an extra weight to sun if the time is dawn/day and the biome has a sunBonus
+    if (applySunBonus) {
+      weatherPool[1] += allBiomes.get(this.biomeType).sunBonus;
+    }
```

Comment:
```suggestion
    const weatherPool = { ...allBiomes.get(this.biomeType).weatherPool };

    // Applies an extra weight to sun if the time is dawn/day and the biome has a sunBonus
    if ([TimeOfDay.DAWN, TimeOfDay.DAY].includes(this.getTimeOfDay())) {
      weatherPool[1] += allBiomes.get(this.biomeType).sunBonus;
    }
```
Since `sunBonus` is 0 by default, and `allBiomes` is defined with `ts-expect-error`, we shouldn't have to check if the sun bonus is defined here.